coqlspclient/proof_state.py

```diff
@@ -318,7 +334,7 @@ def __check_program(self):
         goals = self.coq_file.current_goals()
         if len(goals.program) == 0:
             return
-        id = goals.program[-1][0][1]
+        id = ".".join(self.coq_file.curr_module + [goals.program[-1][0][1]])
```

Comment:
I agree with this change. I suggest a change to `test_obligation.v` to show why this is necessary. The current test file doesn't really test this because the ids are different and in the same module.